percona/controllers/replica/replica.go

```diff
@@ -0,0 +1,320 @@
+		if err != nil {
+			replica := getNewReplicaObject(newCluster, &crv1.Pgreplica{}, i)
+			_, err = clientset.CrunchydataV1().Pgreplicas(newCluster.Namespace).Create(ctx, replica, metav1.CreateOptions{})
+			if err != nil {
+				return errors.Wrapf(err, "create replica %s", replica.Name)
+			}
+			continue
+		}
+		replica := getNewReplicaObject(newCluster, oldReplica, i)
+		if reflect.DeepEqual(newCluster.Spec.PGReplicas.HotStandby, oldCluster.Spec.PGReplicas.HotStandby) {
+			continue
+		}
+
+		replica.ResourceVersion = oldReplica.ResourceVersion
+		replica.Status = oldReplica.Status
+		_, err = clientset.CrunchydataV1().Pgreplicas(newCluster.Namespace).Update(ctx, replica, metav1.UpdateOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "update replica %s", replica.Name)
+		}
+
+	}
+	log.Println("Handle service update")
+	err = createOrUpdateReplicaService(clientset, newCluster)
+	if err != nil {
+		return errors.Wrap(err, "handle replica service")
+	}
+	return nil
+}
+
+func getReplicaName(cluster *crv1.PerconaPGCluster, index int) string {
+	return cluster.Name + "-repl" + strconv.Itoa(index)
+}
+
+func getNewReplicaObject(cluster *crv1.PerconaPGCluster, replica *crv1.Pgreplica, index int) *crv1.Pgreplica {
+	if cluster.Spec.PGReplicas == nil {
+		return nil
+	}
+	labels := make(map[string]string)
+	labels[config.LABEL_PG_CLUSTER] = cluster.Name
+	labels[config.LABEL_NAME] = cluster.Name + "-repl" + strconv.Itoa(index)
+	labels["pgouser"] = "admin"
+	if cluster.Spec.PGReplicas.HotStandby.Labels != nil {
+		for k, v := range cluster.Spec.PGReplicas.HotStandby.Labels {
+			labels[k] = v
+		}
+	}
+	annotations := make(map[string]string)
+	if cluster.Spec.PGReplicas.HotStandby.Annotations != nil {
+		for k, v := range cluster.Spec.PGReplicas.HotStandby.Annotations {
+			annotations[k] = v
+		}
+	}
+	storage := crv1.PgStorageSpec{}
+	if cluster.Spec.PGReplicas.HotStandby.VolumeSpec != nil {
+		storageName := getReplicaName(cluster, index)
+		if len(cluster.Spec.PGReplicas.HotStandby.VolumeSpec.Name) > 0 {
+			storageName = cluster.Spec.PGReplicas.HotStandby.VolumeSpec.Name
+		}
+		storage = crv1.PgStorageSpec{
+			Name:               storageName,
+			StorageClass:       cluster.Spec.PGReplicas.HotStandby.VolumeSpec.StorageClass,
+			AccessMode:         cluster.Spec.PGReplicas.HotStandby.VolumeSpec.AccessMode,
+			Size:               cluster.Spec.PGReplicas.HotStandby.VolumeSpec.Size,
+			StorageType:        cluster.Spec.PGReplicas.HotStandby.VolumeSpec.StorageType,
+			SupplementalGroups: cluster.Spec.PGReplicas.HotStandby.VolumeSpec.SupplementalGroups,
+			MatchLabels:        cluster.Spec.PGReplicas.HotStandby.VolumeSpec.MatchLabels,
+		}
+	}
+
+	replica.ObjectMeta.Name = labels[config.LABEL_NAME]
+	replica.ObjectMeta.Namespace = cluster.Namespace
+	if replica.ObjectMeta.Labels == nil {
+		replica.ObjectMeta.Labels = make(map[string]string)
+	}
+	for k, v := range labels {
+		replica.ObjectMeta.Labels[k] = v
+	}
```

Comment:
```suggestion
	if replica.ObjectMeta.Labels == nil {
		replica.ObjectMeta.Labels = labels
	} else {
	    for k, v := range labels {
		    replica.ObjectMeta.Labels[k] = v
	    }
	}
```

It will save us one map allocation.
But it's worth to implement the same for annotations, and now it's already 2 =)